### PR TITLE
Automatically update tracks.tags with hashtags from the description

### PR DESCRIPTION
**Pull request by oskarrough**
- new `track.tags` text array column
- adds a function `parse_tokens(str, prefix)` that finds "tokens" in a string with a certain prefix, like # or @
- adds a trigger on track insert+update that sets the `tags` by parsing `description`

The parse_tokens function is copy paste from http://shuber.io/parsing-tags-from-text-content-in-postgres/, thanks!

### CODE REVIEW

File: 04-radio4000.sql [hunk 1]
```diff
@@ -82,7 +82,8 @@ create table tracks (
 	url text not null,
 	discogs_url text,
 	title text not null,
-	description text
+	description text,
+	tags text[]  default '{}'
```
Comment (oskarrough) [inline 04-radio4000.sql:86]: not sure if we want this default value? why? Found it on the internet

File: 04-radio4000.sql [hunk 2]
```diff
@@ -135,10 +136,54 @@ $$;
...
+	END;
+$$;
+
+create or replace function parse_track_tags()
+	returns trigger
+	language plpgsql
+as $$
+	begin
+		new.tags = parse_tokens(new.description, '#');
+		-- new.mentions = parse_tokens(new.description, '@');
```
Comment (oskarrough) [inline 04-radio4000.sql:182]: Do we want to add "mentions" right away? Maybe not.

File: 04-radio4000.sql (hunk 1 above)
Comment (user-09876) [inline 04-radio4000.sql:86]: what is the `'{}'` default value?

File: 04-radio4000.sql [hunk 3]
```diff
@@ -135,10 +136,54 @@ $$;
...
+		subquery text;
+		captures text;
+		tokens text[];
+	BEGIN
+		regex := prefix || '(\S+)';
+		matches := 'regexp_matches($1, $2, $3) as captures';
+		subquery := '(SELECT ' || matches || ' ORDER BY captures) as matches';
+		captures := 'array_agg(matches.captures[1])';
+
+		EXECUTE 'SELECT ' || captures || ' FROM ' || subquery
```
Comment (user-09876) [inline 04-radio4000.sql:164]: is this bit safe to let user define what `captures` and `subquery` values are? gotta investigate this bit.

File: 04-radio4000.sql (hunk 1 above)
Comment (oskarrough) [reply 04-radio4000.sql:86]: Copy pasted, let's remove

File: 04-radio4000.sql (hunk 2 above)
Comment (user-09876) [reply 04-radio4000.sql:182]: good question! haven't though of that.
we can have them to try?

File: 04-radio4000.sql (hunk 1 above)
Comment (user-09876) [reply 04-radio4000.sql]: tbh, no idea what it means in term of sql

File: 04-radio4000.sql (hunk 2 above)
Comment (oskarrough) [reply 04-radio4000.sql]: done!

File: 04-radio4000.sql (hunk 3 above)
Comment (oskarrough) [reply 04-radio4000.sql:165]: from a user on Discord:

image

so moving ahead 👍

**Pull request merged**